File: system-test/dotnet/Dsl.Core/UseCase/Configuration.cs
using Dsl.Port;
using Dsl.Core.Shared;

namespace Dsl.Core;

public class Configuration
{
    private readonly string shopUiBaseUrl;
    private readonly string shopApiBaseUrl;
    private readonly string erpBaseUrl;
    private readonly string clockBaseUrl;
    private readonly ExternalSystemMode externalSystemMode;
    private readonly ChannelMode channelMode;

    public Configuration(string shopUiBaseUrl, string shopApiBaseUrl, string erpBaseUrl, string clockBaseUrl,
        ExternalSystemMode externalSystemMode, ChannelMode channelMode = ChannelMode.Dynamic)
    {
        this.shopUiBaseUrl = shopUiBaseUrl;
        this.shopApiBaseUrl = shopApiBaseUrl;
        this.erpBaseUrl = erpBaseUrl;
        this.clockBaseUrl = clockBaseUrl;
        this.externalSystemMode = externalSystemMode;
        this.channelMode = channelMode;
    }

    public string ShopUiBaseUrl => shopUiBaseUrl;
    public string ShopApiBaseUrl => shopApiBaseUrl;
    public string ErpBaseUrl => erpBaseUrl;
    public string ClockBaseUrl => clockBaseUrl;
    public ExternalSystemMode ExternalSystemMode => externalSystemMode;
    public ChannelMode ChannelMode => channelMode;
}

File: system-test/java/src/main/java/com/optivem/shop/systemtest/configuration/Configuration.java
package com.optivem.shop.systemtest.configuration;

import com.optivem.shop.testkit.dsl.port.ChannelMode;
import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public class Configuration {
    private final String shopUiBaseUrl;
    private final String shopApiBaseUrl;
    private final String erpBaseUrl;
    private final String clockBaseUrl;
    private final String taxBaseUrl;
    private final ExternalSystemMode externalSystemMode;
    private final ChannelMode channelMode;
}
